src/openforms/prefill/__init__.py

```diff
@@ -139,11 +145,13 @@ def prefill_variables(submission: "Submission", register=None) -> None:
     state = submission.load_submission_value_variables_state()
     variables_to_prefill = state.get_prefill_variables()
 
-    grouped_fields = defaultdict(list)
+    grouped_fields = defaultdict(dict)
     for variable in variables_to_prefill:
-        grouped_fields[variable.form_variable.prefill_plugin].append(
-            variable.form_variable.prefill_attribute
-        )
+        plugin_id = variable.form_variable.prefill_plugin
+        attribute_name = variable.form_variable.prefill_attribute
+        identifier_role = variable.form_variable.prefill_identifier_role
```

Comment:
```suggestion
        identifier_role = variable.form_variable.prefill_identifier_role or "main"
```
for existing components that don't have this configuration option?

Reply:
the `prefill_identifier_role` attribute on the variable has a default, so I think it should always have a value?

Reply:
you're right, that should work!